lib/alarmist/handler.ex

```diff
@@ -51,7 +51,8 @@ defmodule Alarmist.Handler do
     end
   end
 
-  defp maybe_sleep(_handler, timeout) when is_integer(timeout) and timeout > 0 do
+  defp maybe_sleep(_handler, timeout)
+       when is_integer(timeout) and timeout > 0 and timeout > @handler_retry_interval do
     Process.sleep(@handler_retry_interval)
     timeout - @handler_retry_interval
```

Comment:
How about this?
```suggestion
 defp maybe_sleep(_handler, timeout) when is_integer(timeout) and timeout > 0 do
    sleep_time = min(timeout, @handler_retry_interval)
    Process.sleep(sleep_time)
    timeout - sleep_time
```

Reply:
I like it. Tested and solves the problem we were seeing while also being more clear